start.sh

```diff
@@ -1,29 +1,31 @@
 #!/usr/bin/env bash
 
+# Note: set user using `--env uid=XXXX --env gid=XXXX`, instead of using
+# docker's `--user` flag
+
 if [ ! -s "${HOME}/.conan/registry.txt" ] ; then
     cat "${HOME}/.conan/registry.template.txt" >> "${HOME}/.conan/registry.txt"
 fi
 
 if [ "${uid}" -a "${gid}" ] ; then
     set -e
-    user_name=$(basename ${HOME})
-    if (( 1000 != ${gid} )) ; then
-        groupadd --gid ${gid} ${user_name}
+    if ! grep --quiet ":${gid}:" /etc/group; then
+        groupadd --gid "${gid}" cocaptain
     fi
-    if (( 1000 != ${uid} )) ; then
+    if ! grep --quiet ":x:${uid}:" /etc/passwd; then
         useradd \
             --home-dir "$HOME" \
             --uid ${uid} \
             --gid ${gid} \
             --groups sudo \
-            ${user_name}
+            cocaptain
     fi
     if ((1000 != ${uid} || 1000 != ${gid} )) ; then
         chown ${uid}:${gid} "${HOME}"
         chown ${uid}:${gid} "${HOME}/.ssh"
     fi
     chown ${uid}:${gid} "${HOME}/.conan/registry.txt"
-    su_cmd="sudo --preserve-env --user ${user_name}"
+    su_cmd="sudo --preserve-env --user #${uid} --group #${gid}"
```

Comment:
What's with the `#`?

Reply:
`or a numeric user ID (UID) prefixed with the ‘#’ character (e.g.  #0 for UID 0)` -"man sudo"